requirements.txt

```diff
@@ -1,4 +1,4 @@
 netifaces
 aiohttp
 urllib3
-retry
\ No newline at end of file
+tenacity
```

Comment:
tenacity looks like an awesome tool!

Make sure it is added to the pre-commit file too.

Can we use it for other parts in the code?

Reply:
sorry completly missed these comments. I have added to the pre-commit

Reply:
I havent delved into the code for what else it might be useful for, but there are indeed some cool features there